### DISCUSSION TITLE
Shutter TS130F zigbee & ZBbridge

### DISCUSSION THREAD

**Discussion by alsak0de** [arendst/Tasmota#13003] [Q&A] [answered]
Thanks for reading,

I've recently installed two TS130F-based shutter modules. After a manual ZBbind to the right cluster (0x102, I believe), the devices are perfectly available through ZbSend.

The problem is that the implemented commands are very basic 'open', 'close' and 'stop' and therefore the calibration is not implemented as in zigbee2mqtt implementation. These devices have a default travel time of 10s which is far from sufficient to what I need but there's no way I can change that with ZBbridge unless I pass low level commands,

I could not find any documentation about the low level commands which would allow to do the calibration through ZbSend.

Any clue in how to find the instruction set? Any chance this goes in the fw roadmap eventually?

Thanks!

**Comment by alsak0de** [comment]
I see no answers; wrong place for asking may be?  Thanks

**Reply by ascillato** [reply]
No, it is ok. This is the place. The problem seems that no one here knows about that.

I personally can't help you with that due to I don't have that device. Sorry.

You can also ask in the Tasmota support chat on Discord.

**Reply by barbudor** [reply]
You may have more luck on the official Tasmota Discord server https://discord.gg/Ks2Kzd4

**Reply by s-hadinger** [reply]
Low-level zigbee commands are described here: https://tasmota.github.io/docs/Zigbee/#low-level-commands

Do you know the Zigbee commands to use?

**Reply by alsak0de** [reply]
I'm afraid I don't.  I assumed that because of being sold in different brands, including Tuya and Loratap, it'd be sort of standard.
I couldn't find any documentation, though.
These devices work with zigbee2mqtt so I suppose low level commands must be documented somewhere.

**Comment by s-hadinger** [comment, marked as answer]
Looking at Zigbee2MQTT code, it appears to be in the standard 0x0102 cluster "Window Covering".

And it has `Calibration` boolean id 0xF001 of type enum8 (0x30).

So the command should be 
```
ZbSend {"device":"<device>", "write":{"0102/F001%30":"0x00"}}
or
ZbSend {"device":"<device>", "write":{"0102/F001%30":"0x01"}}
```

It has also `CalibrationTime` is seconds, id 0xF003, type uint16 (0x21).

Example (for 10 seconds)
```
ZbSend {"device":"<device>", "write":{"0102/F003%21":10}}
```

This is all done from code, so I can't verify. If it works, I can add them to the predefined attributes.

https://github.com/Koenkk/zigbee-herdsman/blob/4fed7310d1e1e9d81e5148cf5b4d8ec98d03c687/src/zcl/definition/cluster.ts#L1773

**Reply by alsak0de** [reply]
wow.. that's exactly what I was loooking for! Let me try and report back.
Thanks a lot!

**Reply by alsak0de** [reply]
Just found this [list](https://user-images.githubusercontent.com/2848623/92302837-abbe3880-ef6f-11ea-9f59-9591b7cd885a.png) in case it helps!

I'll let you know whether the calibration process 'works'.

**Reply by alsak0de** [reply]
Hiya... here's some initial feedback.
Calibration command seems to pass along:

22:16:43.520 MQT: tele/SonoffZBGW/RESULT = {"ZbResponse":{"Device":"0xA953","Name":" toldo2","Command":"0102_04","Status":0,"StatusMessage":"SUCCESS","Endpoint":1,"LinkQuality":0}}
22:16:43.907 MQT: tele/SonoffZBGW/_toldo2/SENSOR = {"ZbReceived":{" toldo2":{"Device":"0xA953","Name":" toldo2","TuyaCalibration":1,"Endpoint":1,"LinkQuality":0}}}

But the CalibrationTime setting command isn't:

22:21:06.523 MQT: tele/SonoffZBGW/RESULT = {"ZbResponse":{"Device":"0xA953","Name":" toldo2","Command":"0102_04","Status":141,"StatusMessage":"INVALID_DATA_TYPE","Endpoint":1,"LinkQuality":0}}

It seems to be something about the seconds format. 

Also, I haven't managed to follow the calibration 'algorithm' that works with the wifi+tasmota version.
I'll keep trying and will post again. Cheers

**Reply by alsak0de** [reply]
zbprobe output, in case it helps anyhow:

22:42:38.521 CMD: zbprobe toldo2
22:42:38.528 MQT: stat/SonoffZBGW/RESULT = {"ZbProbe":"Done"}
22:42:38.647 MQT: tele/SonoffZBGW/RESULT = {"ZbPing":{"Device":"0xA953","IEEEAddr":"0x0C4314FFFE82902A","Name":"toldo2"}}
22:42:38.662 MQT: tele/SonoffZBGW/RESULT = {"ZbState":{"Status":32,"ActiveEndpoints":["0x01"]}}
22:42:38.758 MQT: tele/SonoffZBGW/toldo2/SENSOR = {"ZbReceived":{"toldo2":{"Device":"0xA953","Name":"toldo2","ModelId":"TS130F","Manufacturer":"_TZ3000_mowcrwsa","Endpoint":1,"LinkQuality":0}}}
22:42:40.311 MQT: tele/SonoffZBGW/RESULT = {"ZbState":{"Status":33,"Device":"0xA953","Endpoint":"0x01","ProfileId":"0x0104","DeviceId":"0x0202","DeviceVersion":1,"InClusters":["0x0000","0x0004","0x0005","0x0006","0x0102"],"OutClusters":["0x0019","0x000A"]}}
22:42:41.009 ZIG: Zigbee Devices Data saved in EEPROM (448 bytes)
22:42:42.314 ZIG: auto-bind `ZbBind {"Device":"0xA953","Endpoint":1,"Cluster":"0x0006"}`
22:42:42.454 MQT: tele/SonoffZBGW/RESULT = {"ZbBind":{"Device":"0xA953","Name":"toldo2","Status":0,"StatusMessage":"SUCCESS"}}
22:42:44.312 ZIG: auto-bind `ZbBind {"Device":"0xA953","Endpoint":1,"Cluster":"0x0102"}`
22:42:44.459 MQT: tele/SonoffZBGW/RESULT = {"ZbBind":{"Device":"0xA953","Name":"toldo2","Status":0,"StatusMessage":"SUCCESS"}}
22:42:46.319 ZIG: auto-bind `ZbSend {"Device":"0xA953","Config":{"Power":{"MinInterval":1,"MaxInterval":3600}}}`
22:42:46.460 MQT: tele/SonoffZBGW/toldo2/SENSOR = {"ZbReceived":{"toldo2":{"Device":"0xA953","Name":"toldo2","ConfigResponse":{"Power":{"Status":134,"StatusMsg":"UNSUPPORTED_ATTRIBUTE"}},"Endpoint":1,"LinkQuality":5}}}
22:42:48.344 ZIG: auto-bind `ZbSend {"Device":"0xA953","Config":{"CurrentPositionLiftPercentage":{"MinInterval":1,"MaxInterval":14400,"ReportableChange":1}}}`
22:42:48.467 MQT: tele/SonoffZBGW/toldo2/SENSOR = {"ZbReceived":{"toldo2":{"Device":"0xA953","Name":"toldo2","ConfigResponse":{},"Endpoint":1,"LinkQuality":0}}}

**Reply by alsak0de** [reply]
Got it... it was way easier than that. No need to send low code. Turns out all one needs to do to set to, say, 15s is:

{"device":"0xA953","Endpoint":"0x01","write":{"TuyaCalibrationTime":150}}

Thanks @s-hadinger for the investigation and the clues.

**Comment by AF340** [comment]
Hi, 
Using tasmota, di u reach to send actions OPEN/CLOSE/STOP with ZBsend? providing payload with  {state:action} return errors for me. 
(same with position)
Thks

**Comment by alsak0de** [comment]
Yes. Example:

Zbsend { "Device":"0xBE78", "Endpoint":"0x01", "Send":{"ShutterOpen":"" }}

**Reply by AF340** [reply]
Waou, thnks, works fine, I never find any info about shutterOpen/Stop/Close command over the net, congrats :)

**Comment by alsak0de** [comment]
Excellent, I also use this:

{"device":"0xBE78","Endpoint":"0x01","write":{"CurrentPositionLiftPercentage":0}}

to adust daily, as commands will eventually sort of loose track.

Every night, I shut blinds, set lift% to 100 (as if they were fully opened)
and then shut them again so I make sure they’re reset for good

**Reply by AF340** [reply]
Make sense, thanks for the advice :)
